vowpalwabbit/cb_sample.cc

```diff
@@ -93,8 +102,7 @@ base_learner* cb_sample_setup(options_i& options, vw& all)
   if (!cb_sample_option)
     return nullptr;
 
-  data->seed_state = all.random_seed;
-
+  auto data = scoped_calloc_or_throw<cb_sample_data>(all.random_seed);
```

Comment:
Peter's PR https://github.com/VowpalWabbit/vowpal_wabbit/pull/2069 creates a shared random state object.  Perhaps we should be using that here?

Reply:
Yes, I'd love to consume that object here!